### PR TITLE
feat(contracts): move entitlement gated to separate facet in space

### PR DESCRIPTION
**Pull request by giuseppecrj**

### CODE REVIEW

File: contracts/src/spaces/facets/roles/RolesBase.sol [hunk 1]
```diff
@@ -96,42 +96,37 @@ abstract contract RolesBase is IRolesBase {
...
-      rolesWithPermission[i] = Role({
-        id: roleId,
-        name: name,
-        disabled: isImmutable,
-        permissions: permissions,
-        entitlements: entitlements
-      });
+    // Resize the array to remove unused slots
+    assembly {
+      mstore(rolesWithPermission, count)
```
Comment (brianathere) [inline contracts/src/spaces/facets/roles/RolesBase.sol:129]: This is cool!

File: foundry.toml [hunk 2]
```diff
@@ -21,6 +21,7 @@ fs_permissions = [
 ] # Configures permissions for cheatcodes that touch the filesystem
 extra_output = ["metadata", "abi", "bin"]
 extra_output_files = ["metadata", "abi", "bin"]
+ignored_error_codes = [5574]
```
Comment (brianathere) [inline foundry.toml:24]: Why? Comment

**Review by brianathere — APPROVED** [review]
Should this be two PR?

File: foundry.toml (hunk 2 above)
Comment (giuseppecrj) [reply foundry.toml]: removed in latest push, was trying to convert to 0.8.25 but got a bunch of errors

**Review by texuf — APPROVED** [review]

**Pull request merged**